src/rules/orderedImportsRule.ts

```diff
@@ -155,25 +171,118 @@ class OrderedImportsWalker extends Lint.RuleWalker {
         const pair = findUnsortedPair(imports, this.namedImportsOrderTransform);
         if (pair !== null) {
             const [a, b] = pair;
-            this.addFailure(
-                this.createFailure(
-                    a.getStart(),
-                    b.getEnd() - a.getStart(),
-                    Rule.NAMED_IMPORTS_UNORDERED));
+            const start = imports[0].getStart();
+            const end = imports[imports.length - 1].getEnd();
+            const replacement = sortByKey(imports, (x) => this.namedImportsOrderTransform(x.getText())).map((x) => x.getText()).join(", ");
+            this.currentImportsBlock.replaceNamedImports(start, end - start, replacement);
+            this.lastFix = new Lint.Fix(Rule.metadata.ruleName, []);
+            const ruleFailure = this.createFailure(
+                a.getStart(),
+                b.getEnd() - a.getStart(),
+                Rule.NAMED_IMPORTS_UNORDERED,
+                this.lastFix);
+            this.addFailure(ruleFailure);
         }
 
         super.visitNamedImports(node);
     }
 
-    // Check for a blank line, in which case we should reset the import ordering.
+    // keep reading the block of import declarations until the block ends, then replace the entire block
+    // this allows the reorder of named imports to work well with reordering lines
     public visitNode(node: ts.Node) {
         const prefixLength = node.getStart() - node.getFullStart();
         const prefix = node.getFullText().slice(0, prefixLength);
+        const hasBlankLine = prefix.indexOf("\n\n") >= 0 || prefix.indexOf("\r\n\r\n") >= 0;
+        const notImportDeclaration = node.parent != null
+            && node.parent.kind === ts.SyntaxKind.SourceFile
+            && node.kind !== ts.SyntaxKind.ImportDeclaration;
 
-        if (prefix.indexOf("\n\n") >= 0 ||
-            prefix.indexOf("\r\n\r\n") >= 0) {
-            this.lastImportSource = null;
+        if (hasBlankLine || notImportDeclaration) {
+            // end of block
+            if (this.lastFix != null) {
+                const replacement = this.currentImportsBlock.getReplacement();
+                if (replacement != null) {
+                    this.lastFix.replacements.push(replacement);
+                }
+                this.lastFix = null;
+            }
+            this.currentImportsBlock = new ImportsBlock();
         }
         super.visitNode(node);
     }
 }
+
+interface ImportDeclaration {
+    node: ts.ImportDeclaration;
+    nodeStartOffset: number;    // start position of node within source file
+    text: string;
+    sourceFile: string;
```

Comment:
I'd recommend calling this something like `sourcePath` or just `source`, ad it doesn't always refer to a file and `getSourceFile`/`SourceFile` is used by TS with a different meaning in TSLint